include/transport/transport.h

```diff
@@ -3,118 +3,246 @@
 #include <mutex>
 #include <optional>
 #include <queue>
+#include <sys/socket.h>
 #include <vector>
 
 namespace qtransport {
 
-using TransportContextId = uint64_t;
-using MediaStreamId = uint64_t;
+using TransportContextId =
+    uint64_t; ///< Context Id is a 64bit number that is used as a key to maps
+using MediaStreamId = uint64_t; ///< Media stream Id is a 64bit number that is
+                                ///< used as a key to maps
 
-enum class TransportStatus : uint8_t { Ready = 0, Disconnected };
+/**
+ * Transport status/state values
+ */
+enum class TransportStatus : uint8_t {
+  Ready = 0,
+  Connecting,
+  RemoteRequestClose,
+  Disconnected
+};
 
-enum class Error : uint8_t {
+/**
+ * Transport errors
+ */
+enum class TransportError : uint8_t {
   None = 0,
   UnknownError,
   PeerDisconnected,
   PeerUnreachable,
+  CannotResolveHostname,
+  InvalidIpv4Address,
+  InvalidIpv6Address
+};
+
+/**
+ *
+ */
+enum class TransportProtocol { UDP = 0, QUIC };
+
```

Comment:
thinking about it let's make it
`TransportProtocol { UDP=0, PICOQUIC} ` to make it easy to plugin in